### ISSUE TITLE
NewtonSoft.Json 10.0.3 needed, but getting 11.0.0.* - Breaking Durables

### ISSUE THREAD

**Issue by pragnagopa** [Azure/azure-functions-durable-extension#347]
_From @GregLevenhagen on June 13, 2018 20:59_

Yesterday our Durables started producing 500s because of the following NewtonSoft.Json reference on the assembly version. We are not including nor using any NewtonSoft.Json within our own code.

#### Investigative information

- Timestamp: 2018-06-13T20:50:33.457
- Function App version (1.0 or 2.0-beta): 2.0.11857-alpha
- Invocation ID: 71dd8324-4e48-485d-bad1-08078082a159, fbbf76fb-448d-4d2b-84e5-ed17da5d587d
- Region: Central US

#### Repro steps

Run any Durable Function

#### Expected behavior

For the Durable Function to correctly use NewtonSoft.Json 10.0.3

Starting Host (HostId=*-test-funcapp-tripgen, InstanceId=d5048ba1-7a5b-46c1-a22b-e29bb61b9ec2, Version=2.0.11776.0, ProcessId=5340, AppDomainId=1, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=beta) 

#### Actual behavior

Exception while executing function: TripProcessingOrchestrator_HttpStart Could not load type 'Newtonsoft.Json.SerializationBinder' from assembly 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. 

From the Portal Log window

2018-06-13T20:50:33.457 [Error] System.Private.CoreLib: Exception while executing function: TripProcessingOrchestrator_HttpStart. DurableTask.Core: Could not load type 'Newtonsoft.Json.SerializationBinder' from assembly 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
2018-06-13T20:50:33.509 [Error] Executed '*ProcessingOrchestrator_HttpStart' (Failed, Id=fbbf76fb-448d-4d2b-84e5-ed17da5d587d)

#### Known workarounds

No known work-arounds

We've tried manually including the 10.0.3 version of NewtonSoft.Json package.
We've tried manually setting the FUNCTIONS_EXTENSION_VERSION to a previous version, but all combinations of the value for previous versions fail. It detects the custom version, but says it's not valid. What is the format for the value to specify an exact version? I haven't found documentation other than (~1 or beta).

#### Related information 

* Programming language used - C#

_Copied from original issue: Azure/azure-functions-host#2998_

**Comment by pragnagopa** [comment]
_From @GregLevenhagen on June 13, 2018 21:9_

Workaround by setting the following line in the Application Settings
`FUNCTIONS_EXTENSION_VERSION ~2.0.11776`
will show 

>Runtime version loaded from Application Settings: ~2.0.11776
>Warning Unsupported Runtime Version Your custom runtime version is not supported. As a result 2.0.11776.0 runtime is being used. 

but works for now.

**Comment by kashimiz** [comment]
Thanks for this wonderfully thorough issue, Pragna! 😄 this is a known issue with the newest functions host release and we're tracking it here: https://github.com/Azure/azure-functions-durable-extension/issues/342#issuecomment-397488234

Short version: I've got a PR out to update our references but it requires updating some dependencies first. We're working on getting these out.

**Comment by pragnagopa** [comment]
@kashimiz  - @GregLevenhagen filed the original issue. I just moved it to right place :)

**Comment by cgillum** [comment]
This is fixed in the v1.5.0 release.

**Comment by paulceli** [comment]
I updated to 1.5.0. I had the following error while building at first:

```
Version conflict detected for Newtonsoft.Json. Reference the package directly from the project to resolve this issue. 
 sebp.Mobile.TestApp.Func (>= 1.0.0) -> Microsoft.Azure.WebJobs.Extensions.DurableTask (>= 1.5.0) -> DurableTask.AzureStorage.Private (>= 1.2.4) -> Newtonsoft.Json (>= 11.0.2) 
 sebp.Mobile.TestApp.Func (>= 1.0.0) -> Microsoft.NET.Sdk.Functions (>= 1.0.13) -> Newtonsoft.Json (>= 10.0.3).
```

So I added the latest version of Nuget 'Newtonsoft.Json'. Now it builds OK but I get the following errors while starting the app:


```

                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %

Hosting environment: Production
Content root path: /Users/paulceli/Sources/MobileBanking_UITests/sebp.Mobile.TestApp.Func/bin/Debug/netstandard2.0
Now listening on: http://localhost:7071
Application started. Press Ctrl+C to shut down.
[06/22/2018 12:10:39] Reading host configuration file '/Users/paulceli/Sources/MobileBanking_UITests/sebp.Mobile.TestApp.Func/bin/Debug/netstandard2.0/host.json'
[06/22/2018 12:10:39] Host configuration file read:
[06/22/2018 12:10:39] {}
[06/22/2018 12:10:39] Starting Host (HostId=ls888-1606649231, InstanceId=c70bb3a3-06be-436c-8c0d-7c467deea131, Version=2.0.11651.0, ProcessId=2429, AppDomainId=1, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=)
[06/22/2018 12:10:39] Unable to configure java worker. Could not find JAVA_HOME app setting.
[06/22/2018 12:10:39] 
[06/22/2018 12:10:39] Could not configure language worker Java.
[06/22/2018 12:10:39] 
[06/22/2018 12:10:40] Loaded custom extension: DurableTaskExtension from 'referenced by: Method='sebp.Mobile.TestApp.Func.CompareTestRuns.HttpStart', Parameter='starter'.'
Warning: Only got partial types from assembly: Microsoft.Azure.WebJobs.Extensions.DurableTask, Version=1.5.0.0, Culture=neutral, PublicKeyToken=null
Exception message: System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.Assembly.GetTypes()
   at Microsoft.Azure.WebJobs.Host.Indexers.DefaultTypeLocator.FindTypes(Assembly assembly, IEnumerable`1 extensionAssemblies) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Indexers\DefaultTypeLocator.cs:line 120
[06/22/2018 12:10:41] A ScriptHost error has occurred
[06/22/2018 12:10:41] Microsoft.Azure.WebJobs.Extensions.DurableTask: Could not load file or assembly 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. Could not find or load a specific file. (Exception from HRESULT: 0x80131621). System.Private.CoreLib: Could not load file or assembly 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
[06/22/2018 12:10:41] Stopping Host
Function host is not running.
```

😢 
help thanks
